mod/kalvidassign/classes/completion/custom_completion.php

```diff
@@ -0,0 +1,88 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+declare(strict_types = 1);
+
+namespace mod_kalvidassign\completion;
+
+use core_completion\activity_custom_completion;
+
+/**
+ * Activity custom completion subclass for the kalvidassign activity.
+ *
+ * Class for defining mod_kalvidassign's custom completion rules and fetching the completion statuses
+ * of the custom completion rules for a given kalvidassign instance and a user.
+ *
+ * @package mod_kalvidassign
+ * @copyright 2025 Roi Levi <roi.levi@kaltura.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class custom_completion extends activity_custom_completion {
+
+    /**
+     * Fetches the completion state for a given completion rule.
+     *
+     * @param string $rule The completion rule.
+     * @return int The completion state.
+     */
+    public function get_state(string $rule): int {
+        global $DB;
+
+        $this->validate_rule($rule);
```

Comment:
Does the method return something or just throw an exception on validation error?